src/hangar/checkout.py

```diff
@@ -350,10 +636,488 @@ def __repr__(self):
+
+            Unlike the :meth:`__getitem__` method, only ONE of the ``arrayset``
+            name(s) or ``sample`` key(s) can specify multiple elements at the same
+            time. Ie. If multiple ``arraysets`` are specified, only one sample key
+            can be set, likewise if multiple ``samples`` are specified, only one
+            ``arrayset`` can be specified. When specifying multiple ``arraysets``
+            or ``samples``, each data piece to be stored must reside as individual
+            elements (``np.ndarray``) in a List or Tuple. The number of keys and
+            the number of values must match exactally.
+
+        values: Union[np.ndarray, Iterable[np.ndarray]]
+            Data to store in the specified arraysets/sample keys. When
+            specifying multiple ``arraysets`` or ``samples``, each data piece
+            to be stored must reside as individual elements (``np.ndarray``) in
+            a List or Tuple. The number of keys and the number of values must
+            match exactally.
+
+
+        Notes
+        -----
+
+        *  No slicing syntax is supported for either arraysets or samples. This
+           is in order to ensure explicit setting of values in the desired
+           fields/keys
+
+        *  Add multiple samples to multiple arraysets not yet supported.
+
+        """
+        try:
+            tmpconman = not self._is_conman
+            if tmpconman:
+                self.__acquire_writer_lock()
+                self.__enter__()
+
+            if not isinstance(index, (tuple, list)):
+                raise ValueError(f'Idx: {index} does not specify arrayset(s) AND sample(s)')
+            elif len(index) > 2:
+                raise ValueError(f'Index of len > 2 invalid. To multi-set, pass in lists')
+            asetsIdx, sampleNames = index
+
+            # Parse Arraysets
+            if isinstance(asetsIdx, str):
+                asets = [self._arraysets._arraysets[asetsIdx]]
+            elif isinstance(asetsIdx, (tuple, list)):
+                asets = [self._arraysets._arraysets[aidx] for aidx in asetsIdx]
+            else:
+                raise TypeError(f'Arrayset idx: {asetsIdx} of type: {type(asetsIdx)}')
+            nAsets = len(asets)
+
+            # Parse sample names
+            if isinstance(sampleNames, (str, int)):
+                sampleNames = [sampleNames]
+            elif not isinstance(sampleNames, (list, tuple)):
+                raise TypeError(f'Sample names: {sampleNames} type: {type(sampleNames)}')
+            nSamples = len(sampleNames)
+
+            # Verify asets
+            if (nAsets > 1) and (nSamples > 1):
+                raise SyntaxError(
+                    'Not allowed to specify BOTH multiple samples AND multiple'
+                    'arraysets in `set` operation in current Hangar implementation')
+
+            elif (nAsets == 1) and (nSamples == 1):
+                for aset in asets:
```

Comment:
small nit: can't we do it without loops. It looks ugly to do the loop since you are doing the len check.

Reply:
fair enough. fixed!